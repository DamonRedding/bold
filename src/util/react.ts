import { MutableRefObject, Ref } from 'react'

export function composeRefs<T>(...refs: Array<Ref<T>>) {
    return (value: T) => refs.forEach(ref => setRef(ref, value))
}

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}

/**
 * Returns the next sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getNextSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.nextElementSibling) {
        current = current.nextElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}

/**
 * Returns the previous sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getPreviousSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.previousElementSibling) {
        current = current.previousElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}
