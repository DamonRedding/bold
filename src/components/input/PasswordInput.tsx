import * as React from 'react'
import { Input, PublicInputProps } from './Input'

export interface PasswordInputProps extends PublicInputProps {
	placeholder?: string
}

export class PasswordInput extends React.Component<PasswordInputProps, any> {

	render() {
		return (
			<Input {...this.props} type='password' />
		)
	}

}
