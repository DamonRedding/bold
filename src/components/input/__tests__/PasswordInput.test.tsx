import * as React from 'react'
import { shallowRenderAndMatch } from '../../../__tests__/test.defaults'
import { PasswordInput } from '../PasswordInput'

describe('PasswordInput', () => {
    it('render', function () {
        shallowRenderAndMatch(<PasswordInput />)
    })
})
