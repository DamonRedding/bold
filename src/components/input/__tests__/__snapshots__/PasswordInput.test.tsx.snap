// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput render 1`] = `
<Input
  type="password"
/>
`;
